backend/tournesol/serializers/voucher.py

```diff
@@ -0,0 +1,42 @@
+from rest_framework import serializers
+from rest_framework.exceptions import ValidationError
+from rest_framework.serializers import ModelSerializer
+from rest_framework.validators import UniqueTogetherValidator
+
+from core.models import User
+from vouch.models import Voucher
+
+
+class VoucherSerializer(ModelSerializer):
+    by = serializers.HiddenField(default=serializers.CurrentUserDefault())
+    to = serializers.CharField()
+
+    class Meta:
+        model = Voucher
+
+        fields = [
+            "by",
+            "to",
+            "is_public",
+            "value",
+        ]
+
+        validators = [
+            UniqueTogetherValidator(
+                queryset=Voucher.objects.all(),
+                fields=["to", "by"],
+            )
+        ]
```

Comment:
To apply this constraint at the database level, it can be defined in the `Vouch` db model directly:
```py
class Voucher(models.Model):
    class Meta:
        unique_together = ["by", "to"]
```
similarly to what exists in the `RateLater` model for example: https://github.com/tournesol-app/tournesol/blob/5c0ce9b1f64375e686d36b85f4e4b4221631b237/backend/tournesol/models/rate_later.py#L38-L40

(a new [generated migration](https://docs.djangoproject.com/en/4.0/ref/django-admin/#makemigrations) will be required to apply the constraint in the database)

DRF `ModelSerializer` will be able to read the model definition and apply this constraint on validation too, so the `validators` list here can be removed.

Reply:
Ok indeed it works.

When I failed to make it return a proper error it was with this code:

```python
class Voucher(models.Model):
    class Meta:
        constraints = [ models.UniqueConstraint(fields=['by', 'to'], name="unique-voucher-by-to") ]
        indexes = [ models.Index(fields=['by', 'to']) ]
```

I guess in that case the `ModelSerializer` is not able to handle the constraint.